File: src/main/java/eladkay/quaritum/common/item/soulstones/ItemOppressiveSoulstone.java
package eladkay.quaritum.common.item.soulstones;

import eladkay.quaritum.common.item.base.ItemMod;
import eladkay.quaritum.common.lib.LibNames;
import net.minecraft.item.ItemStack;
import net.minecraft.world.World;

/**
 * @author WireSegal
 *         Created at 10:48 AM on 4/30/16.
 */
public class ItemOppressiveSoulstone extends ItemMod {
    public ItemOppressiveSoulstone() {
        super(LibNames.OPPRESSIVE_SOULSTONE);
        setMaxStackSize(1);
    }

    @Override
    public int getEntityLifespan(ItemStack itemStack, World world) {
        return Integer.MAX_VALUE;
    }
}

File: src/main/java/eladkay/quaritum/common/item/soulstones/ItemOppressiveSoulstone.kt
package eladkay.quaritum.common.item.soulstones

import eladkay.quaritum.common.lib.LibNames
import net.minecraft.item.ItemStack
import net.minecraft.world.World

class ItemOppressiveSoulstone : ItemMod(LibNames.OPPRESSIVE_SOULSTONE) {
    init {
        setMaxStackSize(1)
    }

    override fun getEntityLifespan(itemStack: ItemStack?, world: World?): Int {
        return Integer.MAX_VALUE
    }
}
